pkg/resources/shared_database.go

```diff
@@ -0,0 +1,305 @@
+		// TODO(SNOW-1325381)
+		// Transient:                  GetPropertyAsPointer[bool](d, "is_transient"),
+		ExternalVolume:             externalVolume,
+		Catalog:                    catalog,
+		ReplaceInvalidCharacters:   GetPropertyAsPointer[bool](d, "replace_invalid_characters"),
+		DefaultDDLCollation:        GetPropertyAsPointer[string](d, "default_ddl_collation"),
+		StorageSerializationPolicy: storageSerializationPolicy,
+		LogLevel:                   logLevel,
+		TraceLevel:                 traceLevel,
+		Comment:                    GetPropertyAsPointer[string](d, "comment"),
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(helpers.EncodeSnowflakeID(id))
+
+	return ReadSharedDatabase(ctx, d, meta)
+}
+
+func UpdateSharedDatabase(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
+
+	if d.HasChange("name") {
+		newName := sdk.NewAccountObjectIdentifier(d.Get("name").(string))
+		err := client.Databases.Alter(ctx, id, &sdk.AlterDatabaseOptions{
+			NewName: &newName,
+		})
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		d.SetId(helpers.EncodeSnowflakeID(newName))
+		id = newName
+	}
+
+	if d.HasChange("comment") {
+		comment := d.Get("comment").(string)
+		if len(comment) > 0 {
+			err := client.Databases.Alter(ctx, id, &sdk.AlterDatabaseOptions{
+				Set: &sdk.DatabaseSet{
+					Comment: &comment,
+				},
+			})
+			if err != nil {
+				return diag.FromErr(err)
+			}
+		} else {
+			err := client.Databases.Alter(ctx, id, &sdk.AlterDatabaseOptions{
+				Unset: &sdk.DatabaseUnset{
+					Comment: sdk.Bool(true),
+				},
+			})
+			if err != nil {
+				return diag.FromErr(err)
+			}
+		}
+	}
+
+	return ReadSharedDatabase(ctx, d, meta)
+}
+
+func ReadSharedDatabase(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
+
+	database, err := client.Databases.ShowByID(ctx, id)
+	if err != nil {
+		if errors.Is(err, sdk.ErrObjectNotFound) {
+			d.SetId("")
+			return diag.Diagnostics{
+				diag.Diagnostic{
+					Severity: diag.Warning,
+					Summary:  "Failed to query shared database. Marking the resource as removed.",
+					Detail:   fmt.Sprintf("DatabaseName: %s, Err: %s", id.FullyQualifiedName(), err),
+				},
+			}
+		}
+		return diag.FromErr(err)
+	}
+
+	parameters, err := client.Parameters.ShowParameters(ctx, &sdk.ShowParametersOptions{
+		In: &sdk.ParametersIn{
+			Database: id,
+		},
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("name", database.Name); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("from_share", sdk.NewExternalObjectIdentifierFromFullyQualifiedName(database.Origin).FullyQualifiedName()); err != nil {
```

Comment:
Also, wanted to hear your thoughts on setting "name", "from_share" and "as_replica_of" in Read

Reply:
For me it's fine because it is aligned with the Snowflake docs